frontend/src/metabase/lib/dom.js

```diff
@@ -350,12 +367,39 @@ const getLocation = url => {
   try {
     const { pathname, search, hash } = new URL(url, window.location.origin);
     const query = querystring.parse(search.substring(1));
-    return { pathname, search, query, hash };
+    return {
+      pathname: getPathnameWithoutSubPath(pathname),
+      search,
+      query,
+      hash,
+    };
   } catch {
     return {};
   }
 };
 
+function getPathnameWithoutSubPath(pathname) {
+  const pathnameSections = pathname.split("/");
+  const sitePathSections = getSitePath().split("/");
```

Comment:
Both `pathname` and `sitePath` will always have a leading slash, so we don't have to worry about matching leading or nonleading slash.